tf2/src/buffer_core.cpp

```diff
@@ -572,6 +574,123 @@ struct TransformAccum
   tf2::Vector3 result_vec;
 };
 
+geometry_msgs::msg::VelocityStamped BufferCore::lookupVelocity(
+  const std::string & tracking_frame, const std::string & observation_frame,
+  const TimePoint & time, const tf2::Duration & averaging_interval) const
+{
+  // ref point is origin of tracking_frame, ref_frame = obs_frame
+  return lookupVelocity(
+    tracking_frame, observation_frame, observation_frame, tf2::Vector3(
+      0, 0,
+      0), tracking_frame, time,
+    averaging_interval);
+}
+
+geometry_msgs::msg::VelocityStamped BufferCore::lookupVelocity(
```

Comment:
I think that we can simplify this down with our assumptions about points and frames of references being coincident.

Reply:
the method above is a simplified version with some assumptions, can you clarify more how we can simplify this?